components/service/glean/src/main/java/mozilla/components/service/glean/EventMetricType.kt

```diff
@@ -109,4 +111,37 @@ data class EventMetricType(
             )
         }
     }
+
+    /**
+     * Tests whether a value is stored for the metric for testing purposes only
+     *
+     * @return true if metric value exists, otherwise false
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+    fun testHasValue(
+        pingName: String =
+            if (sendInPings[0] == "default") { "metrics" } else { sendInPings[0] }
+    ): Boolean {
+        return EventsStorageEngine.getSnapshot(pingName, false)?.filter {
+            return if (category.isEmpty()) {
+                it.name == getStoredName()
+            } else {
+                "${it.category}.${it.name}" == getStoredName()
+            }
+        } != null
+    }
+
+    /**
+     * Returns the stored value for testing purposes only
+     *
+     * @return value of the stored metric
+     * @throws [NullPointerException] if no value is stored
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+    fun testGetValue(
```

Comment:
Looks good.

I suppose we should `filter` this, too?  And for `has`, I guess we could use `any` since we don't need the actual values there.

Reply:
True, good catch.